From ae9b80797295a654003f628a92e968950ff9689c Mon Sep 17 00:00:00 2001
From: Bruce Ashfield <bruce.ashfield@gmail.com>
Date: Sun, 10 Jul 2022 21:37:07 -0400
Subject: vt/conmakehash: improve reproducibility

The file generated by conmakehash capture the application
path used to generate the file. While that can be informative,
it varies based on where the kernel was built, as the full
path is captured.

We tweak the application to use a second input as the "capture
name", and then modify the Makefile to pass the basename of
the source, making it reproducible.

This could be improved by using some sort of path mapping,
or the application manipualing argv[1] itself, but for now
this solves the reprodicibility issue.

Upstream-Status: Inappropriate [Yocto-specific]
Signed-off-by: Bruce Ashfield <bruce.ashfield@gmail.com>
---
 drivers/tty/vt/Makefile      | 2 +-
 drivers/tty/vt/conmakehash.c | 2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

(limited to 'drivers/tty/vt')

diff --git a/drivers/tty/vt/Makefile b/drivers/tty/vt/Makefile
index b3dfe9d5717e..6ee3736776c8 100644
--- a/drivers/tty/vt/Makefile
+++ b/drivers/tty/vt/Makefile
@@ -15,7 +15,7 @@ clean-files := consolemap_deftbl.c defkeymap.c
 hostprogs += conmakehash
 
 quiet_cmd_conmk = CONMK   $@
-      cmd_conmk = $(obj)/conmakehash $< > $@
+      cmd_conmk = $(obj)/conmakehash $< $(shell basename $<) > $@
 
 $(obj)/consolemap_deftbl.c: $(src)/$(FONTMAPFILE) $(obj)/conmakehash
 	$(call cmd,conmk)
diff --git a/drivers/tty/vt/conmakehash.c b/drivers/tty/vt/conmakehash.c
index cddd789fe46e..d62510b280e9 100644
--- a/drivers/tty/vt/conmakehash.c
+++ b/drivers/tty/vt/conmakehash.c
@@ -253,7 +253,7 @@ int main(int argc, char *argv[])
 #include <linux/types.h>\n\
 \n\
 u8 dfont_unicount[%d] = \n\
-{\n\t", argv[1], fontlen);
+{\n\t", argv[2], fontlen);
 
   for ( i = 0 ; i < fontlen ; i++ )
     {
-- 
cgit 1.2.3-korg

